src/main/java/se/lth/base/server/mail/CustomEmail.java

```diff
@@ -2,64 +2,510 @@
+
+        return EmailBuilder.startingBlank()
+                .from(MailHandler.DEFAULT_SENDER)
+                .to(recipients)
+                .withSubject(subjectAndBody.get("subject"))
+                .withHTMLText(subjectAndBody.get("body"))
+                .withPlainText("Please view this email in a modern email client!")
+                .buildEmail();
+    }
+
+    private Map<String, String> getSubjectAndBody() {
+        String subject = "", body = "";
+        Map<String, String> subjectAndBody = new HashMap<>();
+
+        switch (emailType) {
+            case WELCOME: {
+                subject = getWelcomeSubject();
+                body = getWelcomeBody();
+                break;
+            }
+            case NEW_PASSENGER_ON_TRIP: {
+                subject = getNewPassengerOnTripSubject();
+                body = getNewPassengerOnTripBody();
+                break;
+            }
+            case BOOKING_CONFIRMED: {
+                subject = getBookingConfirmedSubject();
+                body = getBookingConfirmedBody();
+                break;
+            }
+            case RATING: {
+                subject = getRatingSubject();
+                body = getRatingBody();
+                break;
+            }
+            case FILTER_MATCH: {
+                subject = getFilterMatchSubject();
+                body = getFilterMatchBody();
+                break;
+            }
+            case PASSENGER_CANCELLED_TRIP: {
+                subject = getPassengerCancelledTripSubject();
+                body = getPassengerCancelledTripBody();
+                break;
+            }
+            case DRIVER_CANCELLED_DRIVE: {
+                subject = getDriverCancelledDriveSubject();
+                body = getDriverCancelledDriveBody();
+                break;
+            }
+            case WARNING: {
+                subject = getWarningSubject();
+                body = getWarningBody();
+                break;
+            }
+            case DRIVER_REMOVED_PASSENGER: {
+                subject = getDriverRemovedPassengerSubject();
+                body = getDriverRemovedPassengerBody();
+                break;
+            }
+        }
+
+        subjectAndBody.put("subject", subject);
+        subjectAndBody.put("body", body);
+        return subjectAndBody;
+    }
+
+    // Helper methods to fetch HTML email template
+    private String getResource(String resourceName) {
+        InputStream inputStream = CustomEmail.class.getResourceAsStream(resourceName);
+        InputStreamReader streamReader = new InputStreamReader(inputStream);
+
+
+        try (BufferedReader br = new BufferedReader(streamReader)) {
+            return br.lines().collect(Collectors.joining(System.lineSeparator()));
+        } catch (IOException e) {
+            LoggerFactory.getLogger(CustomEmail.class).error(e.getMessage(), e);
+        }
+
+        return "Error: could not read email template";
+    }
```

Comment:
`getResource` is only used in combination with `parseResource`, and vice versa. Combine the two.

Also, returning error as a string is bad practice; throw error instead. As I'm reading it now, we're still sending emails even if we can't read template (?).